app.py

```diff
@@ -200,29 +436,55 @@ def builds(package_name, subdir, version, channel="conda-forge", with_broken=Fal
     ]
 
 
-def extensions(package_name, subdir, version, build, channel="conda-forge", with_broken=False):
+def _extensions_from_anaconda_api(package_name: str, subdir: str, version: str, build: str, channel: str,
+                                  with_broken: bool = False) -> set[str]:
+    data = anaconda_api_data(package_name, channel)
+    return {
+        ("conda" if pkg["basename"].endswith(".conda") else "tar.bz2")
+        for pkg in data
+        if pkg["attrs"]["subdir"] == subdir
+           and pkg["version"] == version
+           and pkg["attrs"]["build"] == build
+           and "main" in pkg["labels"]
+           and (with_broken or "broken" not in pkg["labels"])
+    }
+
+
+def _extensions_from_repodata(package_name: str, subdir: str, version: str, build: str, channel: str) -> set[str]:
+    """
+    with_broken cannot be considered here as repodata does not include yanked packages.
```

Comment:
Some repodatas do have a `removed` section, IIRC.

Reply:
Good point. The problem is that it only contains the artifact filename (e.g., `conda-build-3.28.0-py310h4c7bcd0_0.conda`). So, I could use string operations to extract the version and build number from the artifact filename (is the artifact name standardized?)

WDYT about this?

Reply:
Yep, usually you can assume you can do that:

```python
fn = "conda-build-3.28.0-py310h4c7bcd0_0.conda"
if fn.lower().endswith(".tar.bz2"):
  basename = fn[:len(".tar.bz2")]
  ext = ".tar.bz2"
else:
  basename = fn[:len(".conda")]
  ext = ".conda"
name, version, build = basename.rsplit("-", 2)
```

I think there's some code in the module already doing something like this. Feel free to refactor and reuse.

Reply:
implemented in 20df0169d898ad24279b584ebbbb1d9f4f257da6